Common/UI/Quests/UISelectableQuestStep.cs

```diff
@@ -5,27 +5,58 @@
 namespace PathOfTerraria.Common.UI.Quests;
 
 // ReSharper disable once InconsistentNaming
+
+/// <summary>
+/// Displays a single <see cref="QuestStep"/> for a given quest.<br/>
+/// It'll automatically update as long as it's active, change color to green when done, gray out when locked, and default otherwise.
+/// </summary>
 public class UISelectableQuestStep : UISelectableOutlineRectPanel
 {
-	private QuestStep QuestStep { get; set; }
 	private UISimpleWrappableText Title { get; set; }
+	private Quest Quest => Quest.GetQuest(questName);
+	private QuestStep Step => Quest.QuestSteps[index];
+	
+	private readonly string questName;
+	private readonly int index;
 
-	public UISelectableQuestStep(QuestStep step)
+	public UISelectableQuestStep(int stepIndex, string quest)
 	{
+		questName = quest;
+		index = stepIndex;
+
 		DrawFilled = true;
 		SelectedFillColour = new Color(102, 86, 67) * QuestsUIState.SelectedOpacity;
 		HoverFillColour = new Color(102, 86, 67) * QuestsUIState.HoveredOpacity;
 		Height.Set(22f, 0f);
 		Width.Set(325, 0f);
 
 		// text
-		Title = new UISimpleWrappableText(step.QuestString(), 0.7f);
+		Title = new UISimpleWrappableText(string.Empty, 0.7f);
```

Comment:
Do we even need this line if it's just empty right?

Reply:
Yeah, we still need the text and it's updated later.